graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityBinding.java

```diff
@@ -59,8 +59,5 @@ public void configure(ResourceInfo resourceInfo, FeatureContext context) {
             LOG.debug("Resource method {}#{} requires an authorization checks.", resourceClass.getCanonicalName(), resourceMethod.getName());
             context.register(new ShiroAuthorizationFilter(requiresPermissions));
         }
-
-        // TODO this is the wrong approach, we should have an Environment and proper request wrapping
-        context.register((ContainerResponseFilter) (requestContext, responseContext) -> ThreadContext.unbindSubject());
```

Comment:
hmm, i guess it can't hurt to keep this as well, no?

Reply:
I do not think it makes sense to keep it when there are requests where it is just not executed, in theses cases it provides a false sense of security.

Reply:
fair enough